rest_framework_json_api/views.py

```diff
@@ -98,12 +101,83 @@ def get_queryset(self, *args, **kwargs):
         return qs
 
 
-class ModelViewSet(AutoPrefetchMixin, PrefetchForIncludesHelperMixin, viewsets.ModelViewSet):
+class RelatedMixin(object):
+    """
+    This mixin handles all related entities, whose Serializers are declared in "related_serializers"
+    """
+
+    def retrieve_related(self, request, *args, **kwargs):
+        serializer_kwargs = {}
+        instance = self.get_related_instance()
+
+        if hasattr(instance, 'all'):
+            instance = instance.all()
+
+        if callable(instance):
+            instance = instance()
+
+        if instance is None:
+            return Response(data=None)
+
+        if isinstance(instance, Iterable):
+            serializer_kwargs['many'] = True
+
+        serializer = self.get_serializer(instance, **serializer_kwargs)
+        return Response(serializer.data)
+
+    def get_serializer_class(self):
+        parent_serializer_class = super(RelatedMixin, self).get_serializer_class()
+
+        if 'related_field' in self.kwargs:
+            field_name = self.kwargs['related_field']
+
+            # Try get the class from related_serializers
+            if hasattr(parent_serializer_class, 'related_serializers'):
+                class_str = parent_serializer_class.related_serializers.get(field_name, None)
+                if class_str is None:
+                    raise NotFound
+
+            elif hasattr(parent_serializer_class, 'included_serializers'):
+                class_str = parent_serializer_class.included_serializers.get(field_name, None)
```

Comment:
This doesn't seem to be covered by a test. It would be good to add one as this is also used in a success case.

https://codecov.io/gh/django-json-api/django-rest-framework-json-api/pull/451/diff#D7-140